### PR TITLE
1.12.2 item group api

### PR DESCRIPTION
**Pull request by BoogieMonster1O1**
Ported from 1.16 fabric API

image
image

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.3-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.3.1-bin.zip
```
Comment (halotroop2288) [inline gradle/wrapper/gradle-wrapper.properties:3]: It's about time to upgrade to 7.4.2. But can't we just let @dependabot handle that?

File: legacy-fabric-item-groups-v1/src/main/java/net/legacyfabric/fabric/api/client/itemgroup/FabricItemGroupBuilder.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+	public ItemGroup build() {
+		((ItemGroupExtensions) ItemGroup.BUILDING_BLOCKS).fabric_expandArray();
+		return new ItemGroup(ItemGroup.itemGroups.length - 1, String.format("%s.%s", identifier.getNamespace(), identifier.getPath())) {
+			@Override
+			public ItemStack method_13647() {
+				return stackSupplier.get();
+			}
+
+			@Override
+			public void method_13646(DefaultedList<ItemStack> stacks) {
```
Comment (halotroop2288) [inline legacy-fabric-item-groups-v1/src/main/java/net/legacyfabric/fabric/api/client/itemgroup/FabricItemGroupBuilder.java:124]: I think you should map these before overriding them.

File: legacy-fabric-item-groups-v1/src/main/java/net/legacyfabric/fabric/api/client/itemgroup/FabricItemGroupBuilder.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2020 - 2021 Legacy Fabric
```
Comment (halotroop2288) [inline legacy-fabric-item-groups-v1/src/main/java/net/legacyfabric/fabric/api/client/itemgroup/FabricItemGroupBuilder.java:2]: License version bump. This is a new file, and it's 2022!

File: legacy-fabric-item-groups-v1/src/main/java/net/legacyfabric/fabric/impl/item/group/FabricCreativeGuiComponents.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+import net.minecraft.util.Identifier;
+
+public class FabricCreativeGuiComponents {
+	private static final Identifier BUTTON_TEX = new Identifier("legacy-fabric", "textures/gui/creative_buttons.png");
+	public static final Set<ItemGroup> COMMON_GROUPS = new HashSet<>();
+
+	static {
+		COMMON_GROUPS.add(ItemGroup.SEARCH);
+		COMMON_GROUPS.add(ItemGroup.INVENTORY);
+		COMMON_GROUPS.add(ItemGroup.field_15657);
```
Comment (halotroop2288) [inline legacy-fabric-item-groups-v1/src/main/java/net/legacyfabric/fabric/impl/item/group/FabricCreativeGuiComponents.java:40]: You should map this too. It's `HOTBAR`.

**Review by halotroop2288 — APPROVED** [review]
Looks good, except some minor issues.

**Comment by halotroop2288** [comment]
You should re-target this pull request to https://github.com/Legacy-Fabric/fabric/tree/multiversion now and close #125.

**Comment by arthurbambou** [comment]
Already ported to multiversion branch

**Pull request closed without merging**